src/app/dataset-view/dataset.subscriptions.service.ts

```diff
@@ -14,7 +14,7 @@ import { MaybeNull } from "../common/app.types";
 @Injectable({ providedIn: "root" })
 export class DatasetSubscriptionsService {
     private overview$: Subject<OverviewUpdate> = new ReplaySubject<OverviewUpdate>(1 /*bufferSize*/);
-    private sqlQueryData$: Subject<DataUpdate> = new ReplaySubject<DataUpdate>(1 /*bufferSize*/);
+    private sqlQueryData$: Subject<MaybeNull<DataUpdate>> = new ReplaySubject<MaybeNull<DataUpdate>>(1 /*bufferSize*/);
```

Comment:
I think we should isolate everything about SQL queries from this service. Queries logic should live in a separate module, and shared UI components about queries should also live there.

The queries logic is now supposed to be totally independent on current dataset, unlike other queries about the dataset in this service.

Reply:
Added separate service